### PR TITLE
Add webhook tests and add payload handlers

### PR DESCRIPTION
**Pull request by charlesreid1**
This PR expands on #31 to add webhook tests and payload handlers for those tests.

This fixes up the payload handlers in preparation for testing, and implements payload handler tests.

### CODE REVIEW

**Comment by charlesreid1** [comment]
This is progressing well. We have a basic webhook test and a ping-pong webhook test. Following the todo items here: #33

**Comment by charlesreid1** [comment]
Coming right along. We have two sets of payload tests: (1) tests that check the webhook server is responding correctly, and (2) tests that check that the payloads are being processed correctly.

Set (1) is relatively easy. Set (2) requires more work. The payload handlers for tests will use methods in the GithubTask class to extract info from the payload and print it to the logs. Then the tests check if the correct information was extracted and printed in the logs.

Set (1) is complete and working.

Set (2) is underway, we have one pull request webhook payload test and have added utilities to load and send and assortment of other webhooks. We're going to try and avoid a litany of assertions, if possible.

**Comment by charlesreid1** [comment]
Once we finish with all those tests, we'll be ready to merge this into #31. Then we can move on to writing tests for the DCPPC payload handler.

**Comment by charlesreid1** [comment]
Still working through checklist at #33

**Comment by charlesreid1** [comment]
Wow, that got messy. We ran into some issues with the initialization order. Here's the less fuzzy, more clear construction order.

**The bad way:** The Flask documentation just creates the app in `__init__.py` but this means your Flask app is _created when you compile the library_, not when you import the library. The Flask app is then stored, and is loaded whenever you import the library, in whatever state it was in when you built the library.

If you then try and load a config file for the application in the constructor, it tries to load the config file when you run `python setup.py build install`, which is definitely not what you want.

None of this is what we want.

**The good way:** When the archie library is imported, _no_ objects are created. In particular, no Flask app is created. This means that we can't just say `import archie` and then right away use `archie.webapp.app` to refer to the Flask app - it adds an additional step where we have to create the Flask app. 

```
import archie
app = archie.webapp.create_flask_app()
```

The advantage is, the Flask app is not created (and the config file is not loaded) until the moment that `create_flask_app()` function is run, and no sooner. 

This requires some trickery in the `webapp` submodule - in particular, we have to offer a static function `create_flask_app()` that instantiates our custom Flask object (extended from the Flask base class), then we also have to call (from that function) another function that will initialize all of the routes for our flask app.

**Bottom line:** this was all a blessing in disguise, as it makes the webapp submodule much better organized and clarified the construction order (creating and destroying the Payload Handler, which in turn creates and destroys Tasks, once per payload rather than having them stick around for the entire lifetime of the Flask app).

**Comment by charlesreid1** [comment]
🎉  PR payload tests are working! now we just need to add more tests......

**Comment by charlesreid1** [comment]
next steps:

- merge into `add-tasks` branch
- better organize the payload museum and add additional payloads of interest
- link to all payload types: <https://developer.github.com/v3/activity/events/types/>
- implement tests of DCPPC task functionality (similar pattern, check if right commands are run with right payloads)
- move testing classes out of the module proper, and put them into testing.
